MapboxNavigation/RouteTableViewController.swift

```diff
@@ -68,20 +68,37 @@ class RouteTableViewController: UIViewController {
 extension RouteTableViewController: UITableViewDelegate, UITableViewDataSource {
 
     func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+        return routeController.routeProgress.route.legs.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        // Don't display section header if there is only one step
+        guard routeController.routeProgress.route.legs.count > 1 else {
+            return nil
+        }
+        
+        let sectionName = routeController.routeProgress.route.legs[section].name
+        let legToFrom = sectionName.components(separatedBy: ", ")
```

Comment:
I added using source and destination but I also left this code in place for cases where we do not have source and destination.